app/shared/taginput/taginput.component.ts

```diff
@@ -0,0 +1,156 @@
+﻿import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { User } from '../../core/user/user.model';
+import { UserService } from '../../core/user/user.service';
+
+import {
+  Validators,
+  FormGroup,
+  FormControl
+} from '@angular/forms';
+
+@Component({
+  selector: 'hip-taginput',
+  templateUrl: './app/shared/taginput/taginput.component.html'
+})
+export class TagInputComponent {
+
+  public errorMessage: any;       // Handling error message
+  public names: string[] = [];    // AutoComplete List
+  @Input() role: string;          // User role Passed dynamically
+  @Input() users: number[];         // List of Users updated
```

Comment:
The tagInput component doesn't need to have entire User Array. Since the service call only expects a number array, it is redundant to have User array and moreover if User Array is maintained it is never being used any time, except that IDs are being extracted manually again. So, I don't see any advantage of maintaining consistency with redundancy.